plugin/builtin/replicacounttransformer/ReplicaCountTransformer.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate go run sigs.k8s.io/kustomize/plugin/pluginator
+package main
+
+import (
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/pkg/ifc"
+	"sigs.k8s.io/kustomize/pkg/replica"
+	"sigs.k8s.io/kustomize/pkg/resid"
+	"sigs.k8s.io/kustomize/pkg/resmap"
+	"sigs.k8s.io/kustomize/pkg/transformers/config"
+	"sigs.k8s.io/yaml"
+)
+
+// Find matching replicas declarations and replace the count.
+// Eases the kustomization configuration of replica changes.
+type plugin struct {
+	Replica    replica.Replica    `json:"replica,omitempty" yaml:"replica,omitempty"`
+	FieldSpecs []config.FieldSpec `json:"fieldSpecs,omitempty" yaml:"fieldSpecs,omitempty"`
+}
+
+var KustomizePlugin plugin
+
+func (p *plugin) Config(
+	ldr ifc.Loader, rf *resmap.Factory, c []byte) (err error) {
+
+	p.Replica = replica.Replica{}
+	p.FieldSpecs = nil
+	return yaml.Unmarshal(c, p)
+}
+
+func (p *plugin) Transform(m resmap.ResMap) error {
+	matcher := func(r resid.ResId) bool {
+		return r.ItemId.Name == p.Replica.Name
+	}
+
+	for _, r := range m.GetMatchingIds(matcher) {
+		kMap := m[r].Kunstructured.Map()
+
+		specInterface, ok := kMap["spec"]
```

Comment:
appease my computer science teacher, and make "spec" and "replicas" constants, and use them, e.g.

const (
  fldSpec = "spec"
  fldReplicas = "replicas"
)